corehq/apps/users/util.py

```diff
@@ -45,6 +51,40 @@
 ]
 
 
+def generate_mobile_username(username, domain):
+    """
+    Returns the email formatted mobile username if successfully generated
+    Handles exceptions raised by .validation.validate_mobile_username with user facing messages
+    Any additional validation should live in .validation.validate_mobile_username
+    :param username: required str, expecting the first part of a mobile username
+    :param domain: required str, domain name
+    :return: str, email formatted mobile username
+    Example use: generate_mobile_username('username', 'domain') -> 'username@domain.commcarehq.org'
+    """
+    from .validation import validate_mobile_username
+    error = None
+    try:
+        return validate_mobile_username(username, domain)
+    except InvalidUsernameException:
+        error = _("Username may not contain special characters.")
+        if '..' in username:
+            error = _("Username may not contain consecutive . (period).")
+        elif username.endswith('.'):
+            error = _("Username may not end with a . (period).")
+    except UsernameAlreadyExists as e:
+        if e.is_deleted:
+            error = _("Username '{}' belonged to a user that was deleted and cannot be reused.").format(username)
+        else:
+            error = _("Username '{}' is already taken.").format(username)
+    except ReservedUsernameException:
+        error = _("Username '{}' is reserved.").format(username)
+    except InvalidDomainException:
+        error = _("Domain is required.")
```

Comment:
I don't think this should be handled. The 'domain' argument comes from the URL path so it should never be None. I would suggest not handling this error here at all and letting it cause a 500. If it ever is None then this message won't help the user resolve the issue.

Reply:
That sounds good to me. I struggled to think of a scenario where the domain would be None.